Library/Homebrew/language/node.rb

```diff
@@ -22,8 +22,19 @@ def self.std_npm_install_args(libexec)
       # tell npm to not install .brew_home by adding it to the .npmignore file
       # (or creating a new one if no .npmignore file already exists)
       open(".npmignore", "a") { |f| f.write("\n.brew_home\n") }
+      # npm 5.0.0 broke the assumption Homebrew made that npm would always
+      # create and install real directories when passed a file path. We can
+      # restore the old behaviour by packing the path first with "npm pack"
+      # which also helpfully spits out the filename it just created allowing
+      # us to pass that as the final argument.
+
       # npm install args for global style module format installed into libexec
-      ["--verbose", "--global", "--prefix=#{libexec}", "."]
+      %W[
+        --verbose
+        --global
+        --prefix=#{libexec}
+        #{Dir.pwd}/#{Utils.popen_read("npm pack").chomp}
```

Comment:
Yeah this will work, but it will install BOTH yarn and the module at the current working directory to `buildpath/"vendor/yarn"`which is not what you would expect from it (the installing the module at the cwd too part) and if will only work if the cwd is a valid node module.
It's more like a very dirty hack using the API for something it was not meant to be used to. Supporting such a use case by adding a third `global_npm_install_args` (which does the right thing without installing the cwd as a module too) should be a much cleaner way to do so.